src/Strategy/Scala.hs

```diff
@@ -52,67 +59,79 @@ import Path (
   parseAbsFile,
   toFilePath,
  )
-import Prettyprinter (viaShow)
 import Strategy.Maven.Pom qualified as Pom
-import Strategy.Maven.Pom.Closure (MavenProjectClosure, buildProjectClosures)
-import Strategy.Maven.Pom.Closure qualified as PomClosure
+import Strategy.Maven.Pom.Closure (MavenProjectClosure, buildProjectClosures, closurePath)
+import Strategy.Maven.Pom.PomFile (RawPom (rawPomArtifact, rawPomGroup, rawPomVersion))
 import Strategy.Maven.Pom.Resolver (buildGlobalClosure)
+import Strategy.Scala.Errors (FailedToListProjects (FailedToListProjects), MaybeWithoutDependencyTreeTask (MaybeWithoutDependencyTreeTask))
+import Strategy.Scala.SbtDependencyTree (SbtArtifact (SbtArtifact), analyze, sbtDepTreeCmd)
 import Types (
   DependencyResults (..),
   DiscoveredProject (..),
   DiscoveredProjectType (ScalaProjectType),
-  GraphBreadth (Complete),
+  GraphBreadth (Complete, Partial),
  )
 
 discover ::
   ( Has Exec sig m
   , Has ReadFS sig m
   , Has Diagnostics sig m
+  , Has Logger sig m
   , Has (Reader AllFilters) sig m
   ) =>
   Path Abs Dir ->
   m [DiscoveredProject ScalaProject]
 discover = simpleDiscover findProjects' mkProject ScalaProjectType
   where
-    findProjects' dir = map ScalaProject <$> findProjects dir
+    findProjects' dir = concatMap toScalaProjects <$> (findProjects dir)
 
-newtype ScalaProject = ScalaProject {unScalaProject :: PomClosure.MavenProjectClosure}
+    toScalaProjects :: SbtTargets -> [ScalaProject]
+    toScalaProjects (SbtTargets maybeSbtDepTree closure) = ScalaProject maybeSbtDepTree <$> closure
+
+data ScalaProject = ScalaProject
+  { rawSbtDepTree :: Maybe ByteString
```

Comment:
Yeah, this is unfortunate! But there is no other way without rewriting the entire Scala Analyze module (since we want to consider each sub-project as its own target to stay consistent with previous/current behavior)